### PR TITLE
Master adding canary support

### PR DESCRIPTION
**Pull request by ivannaranjo**
This PR adds support to run the tests against the canary image in Flex. The local `test.sh` script is modified for ease of local run.

The support is added by using an alternate app.yaml, app-canary.yaml, which sets the `beta_settings` so the deployment uses the next canary image.

### CODE REVIEW

File: tools/test.sh [hunk 1]
```diff
@@ -64,8 +64,17 @@ else
     readonly run_script=${workspace}/integration_tests/run_tests.yaml
 fi
 
+# Choose the right .yaml file depending on whether we're deploying against the
+# canary or not.
+if [[ "${USE_FLEX_CANARY:-}" == "1" ]]; then
+    echo "Warning: Deploying using the canaray image."
```
Comment (nkubala) [inline tools/test.sh]: nit: canaray -> canary

File: integration_tests/apps/test-1.0/app-canary.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+runtime: aspnetcore
+env: flex
+beta_settings:
+  image: nextcanary
```
Comment (nkubala) [inline integration_tests/apps/test-1.0/app-canary.yaml]: is this the only required configuration for running these on the nextcanary image? could you link some documentation here (just for my own sake)?

Comment (ivannaranjo) [inline integration_tests/apps/test-1.0/app-canary.yaml]: Yep, I'll add a comment to that effect.

File: tools/test.sh (hunk 1 above)
Comment (ivannaranjo) [inline tools/test.sh]: Done.

**Comment by ivannaranjo** [comment]
Addressed comments, since comments are minor I will merge in.

**Pull request merged**